islandora-indexing-triplestore/src/main/java/ca/islandora/alpaca/indexing/triplestore/processors/ProcessorUtils.java

```diff
@@ -0,0 +1,155 @@
+     */
+    public static String getSubjectUri(final Exchange exchange) throws NoSuchHeaderException {
+        final String uri = exchange.getIn().getHeader(FCREPO_URI, "", String.class);
+        if (uri.isEmpty()) {
+            LOGGER.trace("uri isEmpty");
+            final String base = getMandatoryHeader(exchange, FCREPO_BASE_URL, String.class);
+            final String path = exchange.getIn().getHeader(FCREPO_IDENTIFIER, "", String.class);
+            LOGGER.trace("base is {}, path is {}", base, path);
+            return trimTrailingSlash(base) + path;
+        }
+        LOGGER.trace("Returning {}", uri);
+        return uri;
+    }
+
+    /**
+     * Create a DELETE WHERE { ... } statement from the provided subject
+     *
+     * @param subject the subject of the triples to delete.
+     * @param namedGraph an optional named graph
+     * @return the delete statement
+     */
+    public static String deleteWhere(final String subject, final String namedGraph) {
+        final StringBuilder stmt = new StringBuilder("DELETE WHERE { ");
+
+        if (!namedGraph.isEmpty()) {
+            stmt.append("GRAPH <");
+            stmt.append(encode(namedGraph));
+            stmt.append("> { ");
+        }
+
+        stmt.append("<");
+        stmt.append(encode(subject));
+        stmt.append("> ?p ?o ");
+
+        if (!namedGraph.isEmpty()) {
+            stmt.append("} ");
+        }
+
+        stmt.append('}');
+        return stmt.toString();
+    }
+
+    /**
+     *  Create an INSERT DATA { ... } update query with the provided ntriples
+     *
+     *  @param serializedGraph the triples to insert
+     *  @param namedGraph an optional named graph
+     *  @return the insert statement
+     */
+    public static String insertData(final String serializedGraph, final String namedGraph) {
+        final StringBuilder query = new StringBuilder("INSERT DATA { ");
+
+        if (!namedGraph.isEmpty()) {
+            query.append("GRAPH <");
+            query.append(encode(namedGraph));
+            query.append("> { ");
+        }
+
+        query.append(serializedGraph);
+
+        if (!namedGraph.isEmpty()) {
+            query.append("} ");
+        }
+
+        query.append('}');
+        return query.toString();
+    }
+
+    /**
+     * Tokenize a property placeholder value
+     *
+     * @param context the camel context
+     * @param property the name of the property placeholder
+     * @param token the token used for splitting the value
+     * @return a list of values
+     */
+    public static List<String> tokenizePropertyPlaceholder(final CamelContext context, final String property,
+                                                           final String token) {
+        try {
+            return stream(context.resolvePropertyPlaceholders(property).split(token)).map(String::trim)
+                    .filter(val -> !val.isEmpty()).collect(toList());
+        } catch (final Exception ex) {
```

Comment:
PMD's complaining here:

```
4 	/home/adam/repos/Alpaca/islandora-indexing-triplestore/src/main/java/ca/islandora/alpaca/indexing/triplestore/processors/ProcessorUtils.java 	150 	Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block
```

Linking to https://pmd.github.io/pmd-6.21.0/pmd_rules_java_design.html#avoidcatchinggenericexception